### PR TITLE
DEV-921: add search excerpt for pages that don't have one

### PR DESCRIPTION
**Pull request by carylwyatt**
_(This should be squash-merged... so many commits for so little code 🙈 )_

## The problem

Wordpress has a built-in "excerpt" feature, but Phire used ACF to create our content pages, so a workaround is necessary. While there is an "excerpt" option in the editor, adding an excerpt to all 150+ pages would be time-consuming.

## The fix

Using ACFs "loop," I'm checking for "content" blocks and returning the first 250 characters of the first content block on the page. If that content has heading or links, those will be stripped from the string before being returned.

### CODE REVIEW

File: functions.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ function pg_enqueue_site_files() {
 
 		wp_enqueue_style( 'site-fonts', get_template_directory_uri() . '/fonts.min.css', NULL, filemtime( get_template_directory() . '/fonts.css' ) );
 		if ( 'local' === wp_get_environment_type() ) {
-			wp_enqueue_style( 'firebird-styles', 'https://hathitrust-firebird-common.netlify.app/assets/main.css');
+			wp_enqueue_style( 'firebird-styles', 'https://hathitrust-firebird-common.netlify.app/assets/index.css');
```
Comment (aelkiss) [inline functions.php:41]: Unrelated change?

File: search.php [hunk 2]
```diff
@@ -38,7 +38,51 @@
...
+				// so check that the exceprt exists	   
+				if(get_the_excerpt() ) { 
+				
+				?>
+			   
+				<div><?= the_excerpt(); ?></div>
+				<?php 
+			    } else {  
+			
+				// if there's no excerpt, check the ACF fields for the first "content" block and give me the first 250 words with an ellipses at the end
```
Comment (aelkiss) [inline search.php]: first 250 characters, not words, right?

File: search.php [hunk 3]
```diff
@@ -38,7 +38,51 @@
...
+				if ( have_rows( 'main_blocks' ) ){
```
Comment (aelkiss) [inline search.php:55]: Would it be feasible to extract this to a separate function rather than having it inlined in the search template?

**Review by aelkiss — COMMENTED** [review]
A couple comments/questions, but if we merge it like this I think it's fine -- I don't have a good sense of what's easy or possible in wordpress

File: functions.php (hunk 1 above)
Comment (carylwyatt) [reply functions.php:41]: I originally added the netlify styles/scripts for my local development, but the filename changed while I was out. I guess it's not really related to this ticket, but it needs to be changed in order for my local wordpress theme to function.

File: search.php (hunk 2 above)
Comment (carylwyatt) [reply search.php]: correct, I'll update that

File: search.php (hunk 3 above)
Comment (carylwyatt) [reply search.php:55]: Probably? I suppose I could extract this and put it in `functions.php` then invoke it here. Maybe I should try to emulate what Phire has already done, which is use template pieces and include them in the template. My way is a bit messier, but the Phire way means you have to know where things are in order to change/fix them. It's not really a problem either way, just different ways of doing things.

Comment (carylwyatt) [reply search.php:55]: I looked at phire's template files for examples of what to follow for consistency, and all the template files have bits of the WP loop in them, so I left in all the if/while looping parts, but extracted the function.

Comment (aelkiss) [reply search.php:55]: Looks like a good compromise for now. I would say in general our preferred style for new things we write has been to try to keep as much of the logic separated from the template as possible, but I don't (necessarily) want to impose that on the wordpress theme if there is a kind of typical wordpress way of doing things.

Comment (carylwyatt) [reply search.php:55]: WordPress recently changed how templates work, but Phire (and most of my WP knowledge) uses standards from before those changes. You can see how the official theme from 2021 has the `while` "Loop" in the template files:

https://github.com/WordPress/twentytwentyone/blob/trunk/single.php
https://github.com/WordPress/twentytwentyone/blob/trunk/search.php

It's just how things were done at the time. We could probably work towards updating our templates to reflect the new standards if that's something we want to focus on for the future.
